### DISCUSSION TITLE
Any plans for adding $ref input support to external files?

### DISCUSSION THREAD

**Discussion by dpwdec** [LEGO/AsyncAPI.NET#119] [General]
I was wondering if there's any planned worked for updating `$ref` parsing in spec to read in external files? 

I've been looking through the current code base and can see where this functionality could be added. But I didn't want to start doing the work if its duplicated effort that someone else on the project is already working on.

I think this would be valuable addition as its standard AsyncAPI practice to have a single definition file per service and then for those service files to share message definitions and schemas rather than trying to duplicate and synchronise those definitions with internal `components` object.

Hoping to add support for external files that are stored on the local file system as that is more common I believe rather than doing some sort of fetch to an external repository, however in the future this might also be valuable to make the library fully featured.

**Comment by VisualBean** [comment]
Someone awesome totally added this support already #131
